device_tests/test_minimi_400.py

```diff
@@ -0,0 +1,210 @@
+
+        number_of_tasks = await mower.get_number_of_tasks()
+        logger.debug(f'number of tasks : {number_of_tasks}')
+
+        # TODO: implement get task functioality
+        # for now : don't know how to parse "task number"
+        task_number = 0
+        logger.debug(f'Start requesting task {task_number}')
+        task_response = await mower.get_task(task_number)
+        logger.debug(f'Next task starts at {task_response.next_start_time.strftime("%H:%M:%S")}')
+        logger.debug(f'Next task has a duration of {task_response.duration_in_seconds} seconds')
+        logger.debug(f'Next task must run on MONDAY    {task_response.on_monday}')
+        logger.debug(f'Next task must run on TUESDAY   {task_response.on_tuesday}')
+        logger.debug(f'Next task must run on WEDNESDAY {task_response.on_wednesday}')
+        logger.debug(f'Next task must run on THURSDAT  {task_response.on_thursday}')
+        logger.debug(f'Next task must run on FRIDAY    {task_response.on_friday}')
+        logger.debug(f'Next task must run on SATURDAY  {task_response.on_saturday}')
+        logger.debug(f'Next task must run on SUNDAY    {task_response.on_sunday}')
+        
+        
+        keepalive_response = await mower.send_keepalive()
+
+        mower_state = await mower.mower_state()
+        logger.debug(f'Mower state response {mower_state}')
+
+        mower_activity = await mower.mower_activity()
+        logger.debug(f'Mower activity : {mower_activity}')
+
+        get_mode_response = await mower.get_mode_of_operation()
+        logger.debug(f'Mode : {get_mode_response}')
+
+        next_start_time = await mower.mower_next_start_time()
+        logger.debug(f'Next Start Time : {next_start_time}')
+        if next_start_time:
+            logger.debug("Next start time: " + next_start_time.strftime("%Y-%m-%d %H:%M:%S"))
+        else:
+            logger.debug("No next start time")
+
+        restriction_reason = await mower.get_restriction_reason()
+        logger.debug(f'Restriction reason : {restriction_reason}')
+
+        if must_start:
+            # actually start mowing for 30 minutes
+
+            logger.debug('--------------')
+            logger.debug('start setting mode to manual')
+            await mower.set_mode_of_operation("manual")
+            logger.debug('Mode of operation set to manual')
+            logger.debug('--------------')
+
+
+            logger.debug(f'Overriding mow to 30 mins')
+            await mower.set_mower_override_duration_in_seconds(30*60) # 30 minutes override mow
+            logger.debug(f'override mow finished ')
+            logger.debug('--------------')
+
+
+            start_trigger = await mower.start_trigger_request()
+            logger.debug(f'Start trigger response : {start_trigger}')
+            
+        if must_stop:
+            logger.debug(f'Must stop mowing. Send Park command to mower')
+            await mower.set_mode_of_operation('manual')
```

Comment:
`mower.set_mode_of_operation` takes an `ModeOfOperation` enum, not `str`.

Reply:
changed my code according to comment